src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosResponseProcessor.cs

```diff
@@ -68,9 +68,31 @@ public async Task ProcessException(Exception ex)
                 {
                     throw new Core.Exceptions.RequestEntityTooLargeException();
                 }
-                else if (dce.StatusCode == HttpStatusCode.Forbidden && dce.GetSubStatusValue() == CosmosDbSubStatusValues.CustomerManagedKeyInaccessible)
+                else if (dce.StatusCode == HttpStatusCode.Forbidden)
                 {
-                    throw new Core.Exceptions.CustomerManagedKeyInaccessibleException();
+                    switch (dce.GetSubStatusValue())
+                    {
+                        case CosmosDbSubStatusValues.CmkAadClientCredentialsGrantFailure:
```

Comment:
Since these exceptions all deal with the same underlying component, would it be easier to just have a single exception that takes in the sub-status code and creates an `OperationOutcomeIssue` with the appropriate `Resource` message? The rest of the code then has to handle just this one exception type instead of multiple.

Reply:
That sounds like a great idea, will do.

Reply:
This is way better, thanks!